wallets/index.js

```diff
@@ -106,46 +110,114 @@ export function useWallet (name) {
+  return f.clientOnly || !f.serverOnly
+}
+
+function extractClientConfig (fields, config) {
+  return extractConfig(fields, config, true)
+}
+
+function extractServerConfig (fields, config) {
+  return extractConfig(fields, config, false)
+}
+
 function useConfig (wallet) {
   const me = useMe()
 
   const storageKey = getStorageKey(wallet?.name, me)
-  const [localConfig, setLocalConfig, clearLocalConfig] = useLocalConfig(storageKey)
+  const [clientConfig, setClientConfig, clearClientConfig] = useClientConfig(storageKey)
 
   const [serverConfig, setServerConfig, clearServerConfig] = useServerConfig(wallet)
 
-  const hasLocalConfig = !!wallet?.sendPayment
+  const hasClientConfig = !!wallet?.sendPayment
   const hasServerConfig = !!wallet?.walletType
 
-  const config = {
-    // only include config if it makes sense for this wallet
-    // since server config always returns default values for autowithdraw settings
-    // which might be confusing to have for wallets that don't support autowithdraw
-    ...(hasLocalConfig ? localConfig : {}),
-    ...(hasServerConfig ? serverConfig : {})
+  let config = {}
+  if (hasClientConfig) config = clientConfig
+  if (hasServerConfig) {
+    const { enabled } = config || {}
+    config = {
+      ...config,
+      ...serverConfig
+    }
+    // wallet is enabled if enabled is set in client or server config
+    config.enabled ||= enabled
   }
 
-  const saveConfig = useCallback(async (config) => {
-    if (hasLocalConfig) setLocalConfig(config)
-    if (hasServerConfig) await setServerConfig(config)
-  }, [wallet])
+  const saveConfig = useCallback(async (newConfig, { logger }) => {
+    // NOTE:
+    //   verifying the client/server configuration before saving it
+    //   prevents unsetting just one configuration if both are set.
+    //   This means there is no way of unsetting just one configuration
+    //   since 'detach' detaches both.
+    //   Not optimal UX but the trade-off is saving invalid configurations
+    //   and maybe it's not that big of an issue.
+    if (hasClientConfig) {
+      const newClientConfig = extractClientConfig(wallet.fields, newConfig)
+      await walletValidate(wallet, newClientConfig)
+        .then(() => true)
+        // don't throw on validation errors, only use for control flow
+        .catch(() => false)
+        .then(async (valid) => {
+          if (!valid) return
+          setClientConfig(newClientConfig)
+          logger.ok(wallet.isConfigured ? 'payment details updated' : 'wallet attached for payments')
+          if (newConfig.enabled) wallet.enablePayments()
+          else wallet.disablePayments()
+        })
+    }
+    if (hasServerConfig) {
+      const newServerConfig = extractServerConfig(wallet.fields, newConfig)
+      await walletValidate(wallet, newServerConfig)
+        .then(() => true)
+        .catch(() => false)
+        .then(async (valid) => {
+          if (valid) return await setServerConfig(newServerConfig)
+        })
+    }
```

Comment:
I created `walletValidate` since I needed to reuse the validation logic here to make sure only valid configurations are saved.

Since `fieldValidation` makes sure that at least one configuration (for payments or for receives)  is valid, we don't know if both are valid so we extract the corresponding configuration and validate it independently to see if it's valid before saving it.